src/lib/webSocketClient/WebSocketClientManager.js

```diff
@@ -19,28 +19,48 @@
 
  * ModusBox
  * Vijaya Kumar <vijaya.guthi@modusbox.com> (Original Author)
+ * Shashikant Hirugade <shashikant.hirugade@modusbox.com>
  --------------
  ******/
 
 const WebSocket = require('ws')
 const EventEmitter = require('events')
 class MyEmitter extends EventEmitter {}
-
+const Config = require('../config')
 class WebSocketClientManager {
   constructor (consoleFn) {
     this.ws = {}
+    this.userConfig = {}
     if (consoleFn) {
       this.consoleFn = consoleFn
     } else {
       this.consoleFn = console
     }
   }
 
+  async init () {
+    this.userConfig = await Config.getStoredUserConfig()
+  }
+
   customLog (logMessage) {
     this.consoleFn.log(logMessage)
   }
 
   connect (url, clientName, timeout = 15000) {
+    const tlsOptions = {}
+    const urlObject = new URL(url)
+    if (this.userConfig.CLIENT_MUTUAL_TLS_ENABLED) {
+      const cred = this.userConfig.TLS_CREDS.filter(item => item.HOST === urlObject.host)
+      if (Array.isArray(cred) && cred.length === 1) {
+        this.customLog(`Found the Client certificate for ${urlObject.host}`)
+        tlsOptions.cert = cred[0].CERT
+        tlsOptions.key = cred[0].KEY
+        tlsOptions.rejectUnauthorized = false
```

Comment:
Can you make this default to false even CLIENT_MUTUAL_TLS_ENABLED is not enabled?